stan/math/rev/fun/mdivide_left_spd.hpp

```diff
@@ -283,7 +283,10 @@ inline auto mdivide_left_spd(const T1 &A, const T2 &B) {
     arena_t<ret_type> res = A_llt.solve(arena_B.val());
 
     reverse_pass_callback([arena_A, arena_B, arena_A_llt, res]() mutable {
-      promote_scalar_t<double, T2> adjB = res.adj();
+      using T2_t = std::decay_t<T2>;
+      arena_t<Eigen::Matrix<double, T2_t::RowsAtCompileTime,
+                            T2_t::ColsAtCompileTime>>
```

Comment:
```suggestion
      arena_t<promote_scalar_t<double, T2>>
```
or
```suggestion
     arena_t<decltype(res.adj())>
```

Since `arena_t<>` uses [`plain_type_t` internally](https://github.com/stan-dev/math/blob/80d22fb22525ae8f89a606a0b534aed4db60d401/stan/math/rev/meta/arena_type.hpp#L38) (if I'm understanding that right)